src/linear_codes/mod.rs

```diff
@@ -93,18 +93,21 @@ where
     /// of a polynomial of degree m - 1.
     fn encode(msg: &[F], param: &Self::LinCodePCParams) -> Vec<F>;
 
-    /// Get the representation of the polynomial
-    fn poly_repr(polynomial: &P) -> Vec<F>;
+    /// Represent the polynomial as either coefficients,
+    /// in the univariate case, or evaluations over
+    /// the Boolean hypercube, in the multilinear case.
```

Comment:
I think the trait should be more general than univariate or multilinear only. One could have products of MLs, or even general multilinear polynomials. Imo this should be more along the lines of:
Represent the polynomial as a list of field elements in a suitable format for the PCS of interest - for instance, as a vector of coefficients or evaluations.

Reply:
I'm not sure I agree - we shouldn't try to cover all possible use cases of this trait. Currently there are only ML and univariate and I'm not even sure how a scheme for general multivariate would look like, and if it would even exist in practice?

Reply:
Unfortunately `P` cannot refer to EITHER ML or univariate, that would be ideal.

Reply:
I think we are safe if we assume the user has already encoded their non-ML polynomial to an ML polynomial (if needed), and we explicitly say we support univariate and ML.